test/functional/test_runner.py

```diff
@@ -285,11 +285,13 @@ def main():
 
     # Remove the test cases that the user has explicitly asked to exclude.
     if args.exclude:
-        exclude_tests = [re.sub("\.py$", "", test) + (".py" if ".py" not in test else "") for test in args.exclude.split(',')]
+        exclude_tests = [test.split('.py')[0] for test in args.exclude.split(',')]
```

Comment:
nit: I prefer `test.strip('.py')` just in case there's a very unlikely '.py' in the middle of a file name.

Reply:
Wouldn't work with args?

```
>>> 'foo.py --arg'.strip('.py')
'foo.py --arg'